buffer/src/main/java/io/netty/buffer/FixedCompositeByteBuf.java

```diff
@@ -290,6 +334,18 @@ protected int _getInt(int index) {
     }
 
     @Override
+    protected int _getIntLE(int index) {
+        Component c = findComponent(index);
+        if (index + 4 <= c.endOffset) {
+            return c.buf.getIntLE(index - c.offset);
+        } else if (order() == ByteOrder.BIG_ENDIAN) {
+            return _getShort(index) & 0xFFFF | (_getShort(index + 2) & 0xFFFF) << 16;
+        } else {
+            return (_getShort(index) & 0xffff) << 16 | _getShort(index + 2) & 0xffff;
+        }
+    }
```

Comment:
I could only come up with something like that:

```java
public class CompositeByteBufUtil {

    public static int getInt(ByteBuf buf, int index) {
        if (buf.order() == ByteOrder.BIG_ENDIAN) {
            return (buf.getShort(index) & 0xffff) << 16 | buf.getShort(index + 2) & 0xffff;
        } else {
            return buf.getShort(index) & 0xFFFF | (buf.getShort(index + 2) & 0xFFFF) << 16;
        }
    }

    public static int getIntLE(ByteBuf buf, int index) {
        if (buf.order() == ByteOrder.BIG_ENDIAN) {
            return buf.getShortLE(index) & 0xffff | (buf.getShortLE(index + 2) & 0xffff) << 16;
        } else {
            return (buf.getShortLE(index) & 0xffff) << 16 | buf.getShortLE(index + 2) & 0xffff;
        }
    }

}
```

(which would result into index check when calling next method).

If you think it's a good idea, I can make same change for all data types and push.
Otherwise - the rest of your comments are addressed :)